packages/jasmine/jasmine_node_test.bzl

```diff
@@ -25,7 +25,7 @@ def _js_sources_impl(ctx):
     depsets = []
     for src in ctx.attr.srcs:
         if JSModuleInfo in src:
-            depsets.append(src[JSModuleInfo].sources)
+            depsets.append(src[JSModuleInfo].direct_sources)
```

Comment:
technically this is a breaking change, since users who were relying on nested packages having their tests run will now just get nothing running those tests. I think maybe we have to hold this change for a 4.x branch, unless we want to add some opt-in flag for it.

Reply:
> unless we want to add some opt-in flag for it.
I would prefer this option. Our tests are broken at the moment, so it would be great to get this fix faster than waiting for 4.x

Reply:
I've updated the fix a bit:
1. Added `use_direct_specs` opt-in flag.
2. With this flag, only specs from direct sources of `srcs` will be added to execution. Specs from `deps` no longer taken into account.